### DISCUSSION TITLE
Data Generation from Semiempirical MD

### DISCUSSION THREAD

**Discussion by tomo-oka** [deepmodeling/deepmd-kit#3010] [Q&A]
Dear all,

I am a newbie in this area, and want to ask very basic questions. So basically, I am in the testing phase and I used the water example given from the deepmd-kit. What I did so far: 

1) Generation of MD trajectory using XTB with cp2k. This was done by 64 runs using different seeds. Each run was NVT with 12.42 x12.42x12.42 nm^3 box size at 300 K. It is the example you can find in the cp2k manual. The only difference is that I used XTB instead of DFT. For each run, I simulated 100 ps, of which the first 50 ps was discarded, and each dt = 1 ps was taken. In this manner, I have 51 points for each run, and at the end I have 3264 data points.

Now I have two strategies (Energy unit, force unit have been converted to eV and eV/A):

2.A) I took the coordinates, potential energy and forces, as well as box from the XTB-MD -> train those points. I took 400 data points randomly from the whole set for the validation and the rests are for the training.

2.B) I took the coordinates and box -> calculate single point energy and force by DFT -> train those points. Again 400 data points are taken randomly for the validation.

Finally:

3) After the training with the given example from deepmd-kit using se-e2-a descriptor and 100000 steps, I froze the model and compress the potential.

4) Subsequently put the potential for lammps (as in the example).

After checking the trajectory, to my surprise, the system exploded. 

The questions are:

1) How should the data be generated? Was my approach above completely trash?

2) To my understanding, if the system exploded like this, it might be due to: i) too short training, ii) data sampling is not enough, so that whenever the system explore a new phase space, the system cannot be described.

3) I repeat the above procedure, however instead of XTB, I did classical MD (NPT) with around 800 spc-flex water molecule for 2000ps (with rigid model, it didn't work). I took each 1 ps the coordinates, force, energy and box for the training. After the training the same MD simulation using lammps was done, and the results was all good. 

Best regards,

Tomo Oka

**Comment by njzjz** [comment]
Please take a look at concurrent learning (for example, [DP-GEN](https://github.com/deepmodeling/dpgen)). Our papers describe it:

https://doi.org/10.1016/j.cpc.2020.107206
https://doi.org/10.1063/5.0155600

**Reply by tomo-oka** [reply]
Hi, thanks for the links. I will take a look in detail tomorrow. Indeed, I found today a little mistake in the ordering of my data arrays. I took the xtb trajectory, energies and forces for the training. Subsequently, I did 100ps MD with the potential and the results (RDF and trajectories) were ok. 

However, I noticed a further problem: 
Taking strategy 2.B (as I described above), gave me substantially worse results than strategy 2.A. OH bonds are elongated and the HOH angle are significantly wider. 

Best regards,

Tomo Oka
